modmail/utils/plugins.py

```diff
@@ -0,0 +1,86 @@
+# original source:
+# https://github.com/python-discord/bot/blob/a8869b4d60512b173871c886321b261cbc4acca9/bot/utils/extensions.py
+# MIT License 2021 Python Discord
+"""
+Helper utililites for managing plugins.
+
+TODO: Expand file to download plugins from github and gitlab from a list that is passed.
+"""
+
+
+import glob
+import importlib
+import importlib.util
+import inspect
+import logging
+import typing as t
+from pathlib import Path
+
+from modmail import plugins
+from modmail.log import ModmailLogger
+from modmail.utils.cogs import ExtMetadata
+from modmail.utils.extensions import BOT_MODE, unqualify
+
+log: ModmailLogger = logging.getLogger(__name__)
+
+
+BASE_PATH = Path(plugins.__file__).parent.resolve()
+PLUGIN_MODULE = "modmail.plugins"
+PLUGINS: t.Dict[str, t.Tuple[bool, bool]] = dict()
+
+
+def walk_plugins() -> t.Iterator[t.Tuple[str, bool]]:
+    """Yield plugin names from the modmail.plugins subpackage."""
+    # walk all files in the plugins folder
+    # this is to ensure folder symlinks are supported,
+    # which are important for ease of development.
+    for path in glob.iglob(f"{BASE_PATH}/**/*.py", recursive=True):
```

Comment:
Note in the comment why we're not using Pathlib's glob--doesn't follow symlinks right. Even better, also link the python dev bug report I found for it..

Reply:
That'd be `https://bugs.python.org/issue29475`